File: backend/src/main/java/com/example/backend/model/Vehicle.java
package com.example.backend.model;

import com.example.backend.enums.NotificationStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "vehicles")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String ownerName;

    @Column(nullable = false)
    private String ownerEmail;

    @Column(nullable = false)
    private String vehicleName; // e.g. "Honda Scoopy"

    @Column(nullable = false)
    private Double currentKilometers;

    @Column(nullable = false)
    private Double nextServiceKilometers;

    @Column(nullable = false)
    private Double serviceIntervalKm;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationStatus notificationStatus;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (notificationStatus == null) {
            notificationStatus = NotificationStatus.PENDING;
        }
    }

    @PreUpdate
    public void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/types/vehicle.ts
export type NotificationStatus = 'PENDING' | 'NOTIFIED' | 'ACKNOWLEDGED';

export interface Vehicle {
  id: number;
  ownerName: string;
  ownerEmail: string;
  vehicleName: string;
  currentKilometers: number;
  nextServiceKilometers: number;
  serviceIntervalKm: number;
  kmUntilService: number;
  isDueForService: boolean;
  notificationStatus: NotificationStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateVehicleRequest {
  ownerName: string;
  ownerEmail: string;
  vehicleName: string;
  currentKilometers: number;
}

export interface UpdateKilometersRequest {
  currentKilometers: number;
}

export interface UpdateVehicleRequest {
  vehicleName?: string;
  currentKilometers?: number;
}

export interface AcknowledgeServiceRequest {
  newCurrentKilometers: number;
}
